tensorflow/python/keras/layers/advanced_activations.py

```diff
@@ -333,3 +334,74 @@ def get_config(self):
   @tf_utils.shape_type_conversion
   def compute_output_shape(self, input_shape):
     return input_shape
+
+@keras_export('keras.layers.RReLU')
+class RReLU(Layer):
+  """Randomized Leaky Rectified Liner Unit function
+  It Randomizes the alpha passed to the leaky Relu function
+
+  'f(x) = alpha * x for x <0',
+  'f(x) = x for x>=0'.
+
+  where alpha is randomly sampled from uniform distribution U(lower,upper).
+
+  See: https://arxiv.org/pdf/1505.00853.pdf
+
+  Input shape:
+      Arbitrary. Use the keyword argument `input_shape`
+      (tuple of integers, does not include the samples axis)
+      when using this layer as the first layer in a model.
+
+  Output shape:
+      Same shape as the input.
+
+  Arguments:
+      lower: lower bound of the uniform distribution.
+      upper: upper bound of the uniform distribution.
+      seed: Used to create random seeds.
+  """
+
+  def __init__(self, lower=0.125, upper=0.333, seed=None, training=None, **kwargs):
+    super(RReLU, self).__init__(**kwargs)
+    if lower < 0 or upper < 0  :
+      raise ValueError('lower/upper bound for RandomizedRectifierLayer needs '
+                       'to be > 0. ')
+    if lower > upper :
+      raise ValueError('Upper bound for RandomizedRectifierLayer needs '
+                       'to be higher than lower bound. ')
+
+    self.lower = lower
+    self.upper = upper
+    self.seed =  seed
+    self.training = training
+    if self.training==True:
+       self.alpha = K.random_range(low=lower, high=upper)
+
+  def call(self, inputs):
+
+    def rrelu_inputs_testing():
+       with ops.name_scope("Random_Relu_Testing"):
+            alpha = (self.lower+ self.upper) / 2
+            return K.relu(inputs,alpha=alpha)
+
+    def rrelu_input_training():
+       return K.relu(inputs,alpha=self.alpha)
+
+    with ops.name_scope(self.name, "random-relu", [inputs]):
+       output = tf_utils.smart_cond(self.training,
```

Comment:
Seems like training should be a variable or tensor here, otherwise why use tf.cond? I would guess K.learning_phase is a good start.

Reply:
Yes you are write this is smarter way to do it.

Reply:
@tanzhenyu , thanks alot for the review it has helped alot to improve the quality of the code.I have made all the changes as requested by you,pls check